src/representation.rs

```diff
@@ -37,6 +37,64 @@ pub(crate) trait RepresentationTrait {
     }
 }
 
+impl<'a, const P: usize, const W: usize> core::fmt::Debug for Representation<'a, P, W> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.to_string())
+    }
+}
+
+impl<'a, const P: usize, const W: usize> RepresentationTrait for Representation<'a, P, W> {
+    #[inline]
+    fn insert_encoded_hash(&mut self, h: u32) -> usize {
+        match self {
+            Representation::Small(s) => s.insert_encoded_hash(h) as usize,
+            Representation::Array(a) => a.insert_encoded_hash(h) as usize,
+            Representation::Hll(hll) => hll.insert_encoded_hash(h) as usize,
+        }
+    }
+
+    #[inline]
+    fn estimate(&self) -> usize {
+        match self {
+            Representation::Small(s) => s.estimate(),
+            Representation::Array(a) => a.estimate(),
+            Representation::Hll(hll) => hll.estimate(),
+        }
+    }
+
+    #[inline]
+    fn size_of(&self) -> usize {
+        // A priori, we do not know which of the child structs is the largest and therefore
+        // composing the bulk of the memory usage by the enum. Therefore, we subtract the size of
+        // the variants, and then we sum the size of the enum itself.
```

Comment:
Usually `Representation` enum is not stored directly and has `pub(crate)` visibility. It's only created when accessing `CardinalityEstimator` methods and usually optimized by compiler anyway. What's stored is `CardinalityEstimator` struct and its data.

Reply:
Ok, I see, so basically what I am currently measuring would be the 'runtime peak size' in place of the storage size. I suppose the question then is which one we are more interested in measuring. Which one would you rather go for?

Reply:
We're primarily interested in measuring heap allocated memory usage of overall `CardinalityEstimator` and not the temporary used individual sub-structs `Small`, `Array` and `HyperLogLog`, so that we can reliably and accurately measure how much memory `Vec` or `HashMap` of 1M or 100M of `CardinalityEstimator` uses.

Reply:
`Representation` enum and its sub-structs are never actually stored on the heap, these are stack allocated and quickly disposed structs. I think we should only report how much memory `CardinalityEstimator` uses.